services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/AutoRerouteSpec.groovy

```diff
@@ -67,4 +69,51 @@ class AutoRerouteSpec extends BaseSpecification {
         northboundService.deleteFlow(flow.id)
         //TODO(rtretiak): restore costs that were changed due to portdowns
     }
+
+    def "Flow in 'Down' status tries to reroute when discovering a new ISL"() {
+        given: "Two active switches and flow with one alternate path at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        }
+        assumeTrue("No suiting switches found", srcSwitch && dstSwitch)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 1000
+        northboundService.addFlow(flow)
+        def flowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        when: "Bring all ports down on source switch that are involved in current and alternate paths"
+        List<PathNode> broughtDownPorts = []
+        possibleFlowPaths.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northboundService.portDown(src.switchId, src.portNo)
+        }
+
+        then: "Flow goes to 'Down' status"
+        Wrappers.wait(30) { northboundService.getFlowStatus(flow.id).status == FlowState.DOWN }
+
+        when: "Bring all ports up on source switch that are involved in alternate paths"
+        broughtDownPorts.findAll {
+            it.portNo != flowPath.first().portNo
+        }.each {
+            northboundService.portUp(it.switchId, it.portNo)
+        }
+
+        then: "Flow goes to 'Up' status"
+        Wrappers.wait(30) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Flow was rerouted"
+        def reroutedFlowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        flowPath != reroutedFlowPath
+
+        cleanup: "Bring port involved in original path up and delete flow"
+        northboundService.portUp(flowPath.first().switchId, flowPath.first().portNo)
```

Comment:
Please add `Wrappers.wait(5) {northboundService.getAllLinks().every {it.state != IslChangeType.FAILED}}` after you bring ports back up, since it brings instability to tests that are executed afterwards (spent pretty much time debugging such cases)